connector/dialect.tdd

```diff
@@ -625,6 +625,87 @@
+     <formula>(CASE WHEN (%3 &lt;= 0) THEN (POSITION(%2 IN %1) + 1) WHEN (POSITION(%2 IN SUBSTRING(%1 FROM ((%3)::INT - 1) FOR CHAR_LENGTH(%1))) &gt;= 0) THEN (POSITION(%2 IN SUBSTRING(%1 FROM ((%3)::INT - 1) FOR CHAR_LENGTH(%1))) + (%3)::INT) ELSE 0 END)</formula>
+     <argument type='str' />
+     <argument type='str' />
+     <argument type='real' />
+    </function>
+    <function group='string' name='LEFT' return-type='str'>
+     <formula>(CASE WHEN %2 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1,0,%2) END)</formula>
+     <argument type='str' />
+     <argument type='int' />
+    </function>
+    <function group='string' name='LEFT' return-type='str'>
+     <formula>(CASE WHEN %2 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1,0,(%2)::INT) END)</formula>
+     <argument type='str' />
+     <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %2 &lt;= 0 THEN %1 ELSE SUBSTRING(%1 FROM ((%2) - 1) FOR CHAR_LENGTH(%1)) END)</formula>
+     <argument type='str' />
+     <argument type='int' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %2 &lt;= 0 THEN %1 ELSE SUBSTRING(%1 FROM ((%2)::INT - 1) FOR CHAR_LENGTH(%1)) END)</formula>
+     <argument type='str' />
+     <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %3 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1 FROM ((%2)::INT - 1) FOR %3::INT)  END)</formula>
+     <argument type='str' />
+     <argument type='int' />
+     <argument type='int' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %3 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1 FROM ((%2)::INT - 1) FOR %3::INT)  END)</formula>
+     <argument type='str' />
+     <argument type='real' />
+     <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %3 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1 FROM ((%2) - 1) FOR %3::INT)  END)</formula>
+     <argument type='str' />
+     <argument type='int' />
+     <argument type='real' />
+    </function>
+    <function group='string' name='MID' return-type='str'>
+     <formula>(CASE WHEN %3 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) ELSE SUBSTRING(%1 FROM ((%2)::INT - 1) FOR %3)  END)</formula>
+     <argument type='str' />
+     <argument type='real' />
+     <argument type='int' />
+    </function>
+    <function group='string' name='RIGHT' return-type='str'>
+     <formula>(CASE WHEN %2 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) WHEN %2 &gt;= CHAR_LENGTH(%1) THEN %1 ELSE SUBSTRING(%1 FROM (CHAR_LENGTH(%1) - %2 - 1) FOR %2) END)</formula>
```

Comment:
```suggestion
     <formula>(CASE WHEN %2 &lt;= 0 THEN (CASE WHEN %1 IS NULL THEN NULL ELSE '' END) WHEN %2 &gt;= CHAR_LENGTH(%1) THEN %1 ELSE SUBSTRING(%1 FROM (CHAR_LENGTH(%1) - %2) FOR %2) END)</formula>
```
There may be an off by one error with the 'RIGHT' formula.  
For example if I try `SELECT SUBSTRING('calculation' FROM (CHAR_LENGTH('calculation') - 4 - 1) FOR 4)`
I get `atio` and not the expected `tion`.